modules/exploits/multi/http/gogs_git_hooks_rce.rb

```diff
@@ -319,35 +321,6 @@ def gogs_create_file(last_chunk: false)
     nil
   end
 
-  # Hook the HTTP client method to add specific cookie management logic
-  def send_request_cgi(opts, timeout = 20)
-    res = super
-
-    return unless res
-
-    # HTTP client does not handle cookies with the same name correctly. It adds
-    # them instead of substituing the old value with the new one.
-    unless res.get_cookies.empty?
-      cookie_jar_hash = cookie_jar_to_hash
-      cookies_from_response = cookie_jar_to_hash(res.get_cookies.split(' '))
-      cookie_jar_hash.merge!(cookies_from_response)
-      cookie_jar_updated = cookie_jar_hash.each_with_object(Set.new) do |cookie, set|
-        set << "#{cookie[0]}=#{cookie[1]}"
-      end
-      cookie_jar.clear
-      cookie_jar.merge(cookie_jar_updated)
-    end
-
-    res
-  end
-
-  def cookie_jar_to_hash(jar = cookie_jar)
-    jar.each_with_object({}) do |cookie, cookie_hash|
-      name, value = cookie.split('=')
-      cookie_hash[name] = value
-    end
-  end
-
```

Comment:
cc @cdelafuente-r7 - If there was any other cookie jar workarounds you've implemented the past, let us know and we can check it out :+1:

Reply:
I don't think I've implemented other workaround. Thanks for this great improvement!